hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieWriteHandle.java

```diff
@@ -290,7 +290,8 @@ protected LogFileCreationCallback getLogCreationCallback() {
       @Override
       public boolean preFileCreation(HoodieLogFile logFile) {
         WriteMarkers writeMarkers = WriteMarkersFactory.get(config.getMarkersType(), hoodieTable, instantTime);
-        return writeMarkers.createIfNotExists(partitionPath, logFile.getFileName(), IOType.CREATE,
+        IOType ioType = getHoodieTableMetaClient().getTableConfig().getTableVersion().lesserThan(HoodieTableVersion.EIGHT) ? IOType.APPEND : IOType.CREATE;
+        return writeMarkers.createIfNotExists(partitionPath, logFile.getFileName(), ioType,
```

Comment:
Table version check should be within `WriteMarkers`, not in the write handle.

Reply:
Are you suggesting we also pass if the file being passed in is data file or log file as an additional argument?
we don't have two diff apis to create marker for. base file and log file. 
the api just takes in a file name and IOType. 

so, it makes it difficult. unless we parse the file name and deduce whether its a log file or base file. which the caller already knows it. 

lmk what do you think. I feel it might add unnecessary complication within the impl of createMakers in WriteMarkers.

Reply:
We need a new API for create a maker for a log file without specifying the `ioType` so that the implementation can be different using different implemetation within `WriteMarkers` class instead of adding more complexity in the `HoodieWriteHandle` class.